internal/controller/stages/regular_stages.go

```diff
@@ -1342,6 +1342,7 @@ func (r *RegularStageReconciler) startVerification(
 				exprfn.DataOperations(ctx, r.client, stage.Namespace)...,
 			),
 		},
+		rollouts.WithVars(stage.Spec.Vars),
```

Comment:
There is no need to create a `WithVars`, you can just set `Vars` in `rollouts.WithArgumentEvaluationConfig`.

Reply:
When doing this, I think the type of `Vars` should change to `kargoapi.ExpressionVariable`, and the `ArgumentVariable` can be dropped then. That's what was supposed to happen in my original PR.

Reply:
🤦‍♂️

Reply:
There _is_ a conversion from `[]kargoapi.ExpressionVariable` to `[]rollouts.ArgumentVariable` that needs to happen item-by-item. Does that influence your thinking at all?

Reply:
See my second comment :-).

Reply:
I think they crossed in the mail.

Reply:
Should be all set now.